### ISSUE TITLE
Re-enable load test

### ISSUE THREAD

**Issue by sethvargo** [google-github-actions/upload-cloud-storage#250]
### TL;DR

The current load test can only run 3x in a short period and then we hit quota limits. This makes iterative testing and debugging [extremely painful](https://github.com/google-github-actions/upload-cloud-storage/pull/249#discussion_r846243797), so it's disabled.

### Detailed design

We need to re-enable the load test. Ideas:

- Move to a separate action that are allowed to fail
- Only run on pushes to main (not on PRs)
- Run on a cron schedule instead of on PRs and pushes

There also seems to be something weird with the authentication. The error messages we get are actually 503s from the STS service. It seems like the NPM package for Storage is doing an auth handshake for _each_ file upload. I don't have more advanced telemetry to prove this, but that feels like a bug. The token should be cached and reused for its TTL /cc @bcoe
